website/docs/r/synapse_pipeline.html.markdown

```diff
@@ -0,0 +1,128 @@
+---
+
+# azurerm_synapse_pipeline
+
+Manages a Pipeline inside a Azure Synapse Analytics workspace.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "example-resources"
+  location = "West Europe"
+}
+
+resource "azurerm_storage_account" "example" {
+  name                     = "examplestorageacc"
+  resource_group_name      = azurerm_resource_group.example.name
+  location                 = azurerm_resource_group.example.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+  account_kind             = "StorageV2"
+  is_hns_enabled           = "true"
+}
+
+resource "azurerm_storage_data_lake_gen2_filesystem" "example" {
+  name               = "example"
+  storage_account_id = azurerm_storage_account.example.id
+}
+
+resource "azurerm_synapse_workspace" "example" {
+  name                                 = "example"
+  resource_group_name                  = azurerm_resource_group.example.name
+  location                             = azurerm_resource_group.example.location
+  storage_data_lake_gen2_filesystem_id = azurerm_storage_data_lake_gen2_filesystem.example.id
+  sql_administrator_login              = "sqladminuser"
+  sql_administrator_login_password     = "H@Sh1CoR3!"
+  managed_virtual_network_enabled      = true
+}
+
+resource "azurerm_synapse_firewall_rule" "example" {
+  name                 = "AllowAll"
+  synapse_workspace_id = azurerm_synapse_workspace.test.id
+  start_ip_address     = "0.0.0.0"
+  end_ip_address       = "255.255.255.255"
+}
+
+resource "azurerm_synapse_pipeline" "example" {
+  name                 = "example"
+  synapse_workspace_id = azurerm_synapse_workspace.example.id
+
+  depends_on = [azurerm_synapse_firewall_rule.example]
+}
+```
+
+## Example Usage with Activities
+
+```hcl
+resource "azurerm_synapse_pipeline" "example" {
+  name                 = "example"
+  synapse_workspace_id = azurerm_synapse_workspace.example.id
+  variables = {
+    "bob" = "item1"
+  }
+  activities_json = <<JSON
+[
+	{
+		"name": "Append variable1",
+		"type": "AppendVariable",
+		"dependsOn": [],
+		"userProperties": [],
+		"typeProperties": {
+			"variableName": "bob",
+			"value": "something"
+		}
+	}
+]
+  JSON
+
+  depends_on = [azurerm_synapse_firewall_rule.example]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) Specifies the name of the Synapse Pipeline. Changing this forces a new resource to be created. Must be globally unique.
+
+* `synapse_workspace_id` - (Required) The ID of the Synapse Workspace on which to create the Synapse Pipeline. Changing this forces a new resource to be created.
+
+* `description` - (Optional) The description for the Synapse Pipeline.
+
+* `annotations` - (Optional) List of tags that can be used for describing the Synapse Pipeline.
+
+* `parameters` - (Optional) A map of parameters to associate with the Synapse Pipeline.
+
```

Comment:
Does it make sense to mention that currently we only support type string of parameters here?

Reply:
We are trying to keep align with data factory [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/b0c897055329438be6a3a159f6ffac4e1ce958f2/website/docs/r/data_factory_pipeline.html.markdown)

Reply:
Data Factory's docs should be updated to account for that too tbh

Reply:
updated.